VERSIONING.md

```diff
@@ -1,13 +1,40 @@
-# Versioning
+# Versioning and stability
 
-This document describes the versioning policy for this repository. This policy
-is designed so that the following goal can be achieved:
+The OpenTelemetry Collector SIG produces several artifacts for [a variety of audiences](CONTRIBUTING.md#target-audiences). This document describes the versioning and support policy for these artifacts. These policies are designed so that the following goal can be achieved:
 
-**Users are provided a codebase of value that is stable and secure.**
+**Users are provided software artifacts of value that are stable and secure.**
 
-## Public API expectations
+The policies are divided depending on the artifact's target audience. While an artifact is supported, [critical bugs](docs/release.md#bugfix-release-criteria) and security vulnerabilities MUST be addressed. The main criteria for the length of support for an artifact is how easy it is for an artifact's target audience to adapt to disruptive changes.
 
-The following public API expectations apply to all modules in opentelemetry-collector and opentelemetry-collector-contrib.
+## End-user focused software artifacts
+
+End-user focused software artifacts are those that are intended to be used by [end-users](CONTRIBUTING.md#end-users) of the OpenTelemetry Collector. These artifacts include 
+- Binary distributions of the OpenTelemetry Collector.
+- Collector components, such as receivers, processors, and exporters.
+
+These artifacts are versioned according to the [semantic versioning v2.0.0](https://semver.org/) specification.
+
+### General considerations
+
+Binary distributions produced by the Collector SIG contain components and features with varying [levels of stability](README.md#stability-levels). We abide by the following principles to relate the Collector's version to the stability of its components and features:
+
+* The Collector's core framework behavior MUST be stable in order for a Collector distribution to be v1.0.0 or higher.
+* Users can easily understand when they are opting in to use a component or feature that is not stable.
+    * The Collector MUST be configurable in a way that does not use unstable components or features.
+    * The Collector's telemetry (e.g. Collector logs) MUST provide the ability to identify usage of unstable components or features.
+
+### Long term support after v1
+
+The OpenTelemetry Collector SIG provides long term support for stable binary distributions of the OpenTelemetry Collector and its components. The following policies apply to long term support for any major version starting on v1:
+
+* A binary distribution of the OpenTelemetry Collector MUST be supported for a minimum of **one year** after the release of the next major version of said distribution.
```

Comment:
It would be good to have a section in this doc specifically defining what "supported" means

Reply:
When does support for a binary end?

Reply:
> It would be good to have a section in this doc specifically defining what "supported" means

The intention was that this sentence above:

> [critical bugs](docs/release.md#bugfix-release-criteria) and security vulnerabilities MUST be addressed

was the definition of supported (or, at least, a minimum criteria for what supported means). We may do more than this, but that's the minimum.

> When does support for a binary end?

This was also meant to be already covered in the text by this:

> A binary distribution of the OpenTelemetry Collector MUST be supported for a minimum of **one year** after the release of the next major version of said distribution.

So, basically, we support the distro indefinitely until we release the next major version, after which we support it for one more year.

Reply:
@codeboten if you think we should file an issue to continue discussing this please do so!